### DISCUSSION TITLE
string support for $tmpl (template)

### DISCUSSION THREAD

**Discussion by Hunam6** [vlang/v#11879] [Ideas]
Right now only string literals are supported, string support too would be awesome!

**Example of use:**

A dynamic HTML template renderer
```v
page := 'index'
println($tmpl('/templates/${page}.html'))
```

**What did you expect to see?**
The rendered HTML file printed.
 
**What did you see instead?**
![image](https://user-images.githubusercontent.com/38606542/122992037-fa757b80-d3a5-11eb-9d06-099b1321a30f.png)

**Comment by crthpl** [comment]
The question is, how does V know at compile time what the value of `page` is? It is simple for something like your example, but for more complex strings or values, e.g. something calculated from a for loop would V have to run that loop *at compile time*. What if that loop takes hours to complete? What if the environment is different at compile time then when you run it?

**Comment by Nukiloco** [comment]
> 
> 
> The question is, how does V know at compile time what the value of `page` is? It is simple for something like your example, but for more complex strings or values, e.g. something calculated from a for loop would V have to run that loop _at compile time_. What if that loop takes hours to complete? What if the environment is different at compile time then when you run it?

Isn't Vlang strictly static typed? Doesn't that mean Vlang always know what type a value is going to be? Why would a loop taking hours to complete relate to this issue to printing a variable in a string? Also how does a different environment causes the type to completely change types unless if I'm missing the point here?

**Comment by shadowninja55** [comment]
> > The question is, how does V know at compile time what the value of `page` is? It is simple for something like your example, but for more complex strings or values, e.g. something calculated from a for loop would V have to run that loop _at compile time_. What if that loop takes hours to complete? What if the environment is different at compile time then when you run it?
> 
> Isn't Vlang strictly static typed? Doesn't that mean Vlang always know what type a value is going to be? Why would a loop taking hours to complete relate to this issue to printing a variable in a string? Also how does a different environment causes the type to completely change types unless if I'm missing the point here?

the value of `page` is not guaranteed to be known at compile-time while the value of the argument of `$tmpl` has to be known at compile time. we could possibly look into allowing this with constants however.
